Homework2/task8/src/homework/commands/CommandPerformer.java

```diff
@@ -0,0 +1,33 @@
+package homework.commands;
+
+import homework.*;
+
+import java.util.*;
+
+/**
+ * Class CommandPerformer performs available commands from user.
+ */
+public class CommandPerformer {
+
+    /**
+     * Creates an object of CommandPerformer, gets list of product, list of commands
+     * and performs commands, got from user.
+     *
+     * @param products list of created products.
+     * @param commands list of commands.
+     */
+    public CommandPerformer(ArrayList<Product> products, ArrayList<Command> commands) {
+        Scanner scanner = new Scanner(System.in);
+        String line;
+        do {
+            System.out.println("Set command (input 'exit' to exit):");
+            line = scanner.nextLine();
+            // Calls classes of commands to check input command
+            for (Command c : commands) {
+                c.checkCommand(line, products);
```

Comment:
checkCommand is confusing name - it's not obvious that it not only checks that command could be applied but also perform action by itself. It's better to split for two methods - canExecute() and execute() or something like that. Using canExecute also makes command name validation possible - if no c can execute user command then it's possibly due to the invalid input command name